src/index.ts

```diff
@@ -27,81 +28,59 @@ if ('serviceWorker' in navigator && process.env.NODE_ENV !== 'development') {
-  dbOpenRequest.addEventListener('error', (e) => {
-    console.log('error loading db:', e);
-    // TODO: display error
-  });
-  dbOpenRequest.addEventListener('upgradeneeded', (e) => {
-    console.log(`DB update request:`, e);
-    (e.target as IDBRequest).result.createObjectStore(objStoreName);
-  });
+  const dbPromise = storage.LlamaStorage.create();
+  dbPromise.then((db) => onDBOpenSuccess(db, imgElement, shareBtn));
 
-  fileInput.addEventListener('fileselected', (e) => onFileInputChange(e as CustomEvent, dbOpenRequest, imgElement, shareBtn));
+  fileInput.addEventListener('fileselected', (e) => onFileInputChange(e as CustomEvent, dbPromise, imgElement, shareBtn));
 
   shareBtn.addEventListener('click', async () => {
-    const blob = await fetch(imgElement.src).then(r => r.blob());
-    const f = fileFromBlob(blob);
+    const f = await fileFromID(await dbPromise, storage.mainImageName /* TODO: support multiple */);
     navigator.share({files: [f]});
   });
 });
 
-function onDBOpenSuccess(e: Event, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
-  console.log('Database initialized');
-
-  const db = (e.target as IDBRequest).result;
-  const t = db.transaction(objStoreName, 'readonly').objectStore(objStoreName).get(mainImageName);
-  t.addEventListener('success', (e: Event) => {
-    console.log('get success:', e)
-    const b = (e.target as IDBRequest).result;
-    if (!b) return;
+async function onDBOpenSuccess(db: storage.LlamaStorage, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
+  try {
+    const b = await db.getFile(storage.mainImageName /* TODO: support multiple */);
     imgElement.src = window.URL.createObjectURL(b);
-    displayIfShareEnabled(shareBtn, b);
-  });
-  t.addEventListener('error', (e: Event) => {
-    console.log('get error:', e)
-  });
+    displayIfShareEnabled(shareBtn, db);
+  } catch {}
 }
 
-async function onFileInputChange(e: CustomEvent, dbOpenRequest: IDBOpenDBRequest, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
+async function onFileInputChange(e: CustomEvent, dbPromise: Promise<storage.LlamaStorage>, imgElement: HTMLImageElement, shareBtn: HTMLButtonElement) {
+  const db = await dbPromise;
   const f = e.detail as File;
   const b = new Blob([await f.arrayBuffer()], { type: f.type });
   // TODO: perhaps prompt before silently replacing old image, if one exists?
   imgElement.src = window.URL.createObjectURL(b);
-  displayIfShareEnabled(shareBtn, b);
-  const t = dbOpenRequest.result.transaction(objStoreName, 'readwrite').objectStore(objStoreName).put(b, mainImageName);
-  // TODO: display "saving..." message/spinner?
-  t.addEventListener('success', (e) => {
-    console.log('put success:', e)
-  });
-  t.addEventListener('error', (e) => {
-    console.log('put error:', e)
-    // TODO: display error
-  });
+  const recordReady = db.add(b, {
+    filename: f.name,
+    mimeType: f.type,
+    // title: '',
```

Comment:
nit: s/b/blob/
nit: s/f/file/

We're using the full names on :82 and :83, and while I indeed prefer descriptive variable names, we should at the very least be consistent at the file-level until we figure this out through linting.

Reply:
done.